lib/3scale/backend/transactor.rb

```diff
@@ -182,6 +196,24 @@ def preload_usage_limits(application, no_body_enabled, usage_in_params)
         end
       end
 
+      def filter_usages_with_0(transactions)
+        res = transactions.clone
+
+        # There are plenty of existing tests using both a string and a symbol
+        # when accessing the usage.
+        usage_key = ['usage'.freeze, :usage]
+
+        res.each do |_idx, tx|
+          usage_key.each { |k| tx[k]&.reject! { |_metric, delta| delta.to_s == '0'.freeze } }
+        end
+
+        res.reject! do |_idx, tx|
+          usage_key.all? { |k| tx[k].nil? || tx[k].empty? }
+        end
+
+        res
```

Comment:
This looks like it works a bit too much - how about this other version walking the txns and doing lookups just once?
```suggestion
        res.delete_if do |_idx, tx|
          usage = tx['usage'.freeze] || tx[:usage] or next
          usage.delete_if { |_metric, delta| delta.to_s == '0'.freeze }.empty?
        end
```

Reply:
Done. Check new commit.